src/actions-util.ts

```
@@ -161,6 +162,147 @@ export const determineBaseBranchHeadCommitOid = async function (
   }
 };
 
+/**
+ * Deepen the git history of the given ref by one level. Errors are ignored.
+ */
+export const deepenGitHistory = async function () {
```

Two comments about the additions to this file in general:

- Why completely ignore errors?  Does it make sense to at least debug log them?
- Nit, optional: This PR adds a bunch of functionality that interacts with git — now might be a good time to pull this functionality out into its own file.

---

The error is indeed logged centrally within `runGitCommand()`.  By "Errors are ignored", I was trying to convey that errors are not reported back to the caller.  I have update the function documentation to clarify the behavior.

Moving the git interactions to its own file sounds like a good idea, though trying to incorporate that into this PR is a bit messy.  Can I defer that to a dedicated PR after this one merges?

---

Yes, that's fine!